mimic/test/test_core.py

```diff
@@ -92,3 +102,236 @@ def cleanup():
+        self.assertIsNotNone(iapi)
+        helper = APIMockHelper(self, [iapi])
+        core = helper.core
+
+        service_id = None
+        for a_service_id in core._uuid_to_api_internal:
+            if core._uuid_to_api_internal[a_service_id] == iapi:
+                service_id = a_service_id
+
+        resource = core.service_with_region(
+            u"ORD",
+            service_id,
+            u"http://some/random/prefix"
+        )
+        self.assertTrue(
+            IResource.providedBy(resource)
+        )
+
+    def test_uri_for_service(self):
+        """
+        Validate that the URI returned by uri_for_service
+        returns the appropriate base URI
+        """
+        iapi = ExampleAPI()
+        self.assertIsNotNone(iapi)
+        helper = APIMockHelper(self, [iapi])
+        core = helper.core
+
+        service_id = None
+        for a_service_id in core._uuid_to_api_internal:
+            if core._uuid_to_api_internal[a_service_id] == iapi:
+                service_id = a_service_id
+
+        base_uri = "http://some/random/prefix"
+
+        uri = core.uri_for_service(
+            u"ORD",
+            service_id,
+            base_uri
+        )
+        self.assertTrue(
+            uri.startswith(base_uri + '/')
+        )
+
+    def test_entries_for_tenant_external(self):
+        """
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+
+        core = MimicCore(Clock(), [eeapi])
+
+        prefix_map = {}
+        base_uri = "http://some/random/prefix"
+        catalog_entries = [
+            entry
+            for entry in core.entries_for_tenant(
+                'some-tenant',
+                prefix_map,
+                base_uri
+            )
+        ]
+
+        self.assertEqual(len(catalog_entries), 1)
+
+    def test_entries_for_tenant_internal(self):
+        """
+        """
+        core = MimicCore(Clock(), [ExampleAPI()])
+
+        prefix_map = {}
+        base_uri = "http://some/random/prefix"
+        catalog_entries = [
+            entry
+            for entry in core.entries_for_tenant(
+                'some-tenant',
+                prefix_map,
+                base_uri
+            )
+        ]
+
+        self.assertEqual(len(catalog_entries), 1)
+
+    def test_entries_for_tenant_combined(self):
+        """
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+
+        core = MimicCore(Clock(), [eeapi, ExampleAPI()])
+
+        prefix_map = {}
+        base_uri = "http://some/random/prefix"
+        catalog_entries = [
+            entry
+            for entry in core.entries_for_tenant(
+                'some-tenant',
+                prefix_map,
+                base_uri
+            )
+        ]
+
+        self.assertEqual(len(catalog_entries), 2)
+
+    def test_entries_for_tenant_external_invalid(self):
```

Comment:
So this test was my attempt at trying to testing [this little block of code](https://github.com/BenjamenMeyer/mimic/blob/enhancement_iexternalapitemplatemock/mimic/core.py#L196), though unsuccessfully. See [this comment](https://github.com/rackerlabs/mimic/pull/614#discussion_r62910112) for details. I'm all :ear:'s for solutions.